packages/material-ui-lab/src/ClockPicker/ClockPicker.test.tsx

```diff
@@ -21,4 +29,172 @@ describe('<ClockPicker />', () => {
+      render(
+        <ClockPicker
+          ampm={false}
+          date={adapterToUse.date('2018-01-01T13:00:00.000')}
+          minTime={adapterToUse.date('2018-01-01T12:15:00.000')}
+          maxTime={adapterToUse.date('2018-01-01T15:45:30.000')}
+          onChange={handleChange}
+          onViewChange={handleViewChange}
+          view="minutes"
+        />,
+      );
+
+      fireTouchChangedEvent(getByMuiTest('clock'), 'touchmove', clockTouchEvent['--:20']);
+
+      expect(handleChange.callCount).to.equal(1);
+      const [date, selectionState] = handleChange.firstCall.args;
+      expect(date).toEqualDateTime(adapterToUse.date('2018-01-01T13:20:00.000'));
+      expect(selectionState).to.equal('shallow');
+      expect(handleViewChange.callCount).to.equal(0);
+    });
+
+    it('should not select minute when hour is disabled', () => {
+      const handleChange = spy();
+      render(
+        <ClockPicker
+          ampm={false}
+          date={adapterToUse.date('2018-01-01T20:00:00.000')}
+          minTime={adapterToUse.date('2018-01-01T12:15:00.000')}
+          maxTime={adapterToUse.date('2018-01-01T15:45:30.000')}
+          onChange={handleChange}
+          view="minutes"
+        />,
+      );
+
+      fireTouchChangedEvent(getByMuiTest('clock'), 'touchmove', clockTouchEvent['--:20']);
+
+      expect(handleChange.callCount).to.equal(0);
+    });
+
+    it('should not select disabled hour', () => {
+      const handleChange = spy();
+      render(
+        <ClockPicker
+          ampm={false}
+          date={adapterToUse.date('2018-01-01T13:00:00.000')}
+          minTime={adapterToUse.date('2018-01-01T12:15:00.000')}
+          maxTime={adapterToUse.date('2018-01-01T15:45:30.000')}
+          onChange={handleChange}
+          view="hours"
+        />,
+      );
+
+      fireTouchChangedEvent(getByMuiTest('clock'), 'touchmove', clockTouchEvent['20:--']);
+
+      expect(handleChange.callCount).to.equal(0);
+    });
+
+    it('should select enabled second', () => {
+      const handleChange = spy();
+      const handleViewChange = spy();
+      render(
+        <ClockPicker
+          ampm={false}
+          date={adapterToUse.date('2018-01-01T13:20:00.000')}
+          minTime={adapterToUse.date('2018-01-01T12:15:00.000')}
+          maxTime={adapterToUse.date('2018-01-01T15:45:30.000')}
+          onChange={handleChange}
+          onViewChange={handleViewChange}
+          view="seconds"
+        />,
+      );
+
+      fireTouchChangedEvent(getByMuiTest('clock'), 'touchmove', clockTouchEvent['--:10']);
+
+      expect(handleChange.callCount).to.equal(1);
+      const [date, selectionState] = handleChange.firstCall.args;
+      expect(date).toEqualDateTime(adapterToUse.date('2018-01-01T13:20:10.000'));
+      expect(selectionState).to.equal('shallow');
+      expect(handleViewChange.callCount).to.equal(0);
+    });
+
+    it('should not select second when hour is disabled', () => {
```

Comment:
This was previously worded "should not select disabled second" which is inaccurate. The reason the change wasn't fired was that the hour is disabled. So I used a similar wording from a previous test ("should not select minute when hour is disabled").

Now that I think about it: This wording make it sound like each time part is viewed differently  i.e. we're validating each part individually. But we actually check if the full time is within `[minTime, maxTime]` not just if the selected hour is in `[getHour(minTime), getHour(maxTime)]`.

So 
```suggestion
    it('should not select second when time is disabled', () => {
```
sounds less ambigious. At least I was confused about this originally thinking that selecting 30s wouldn't be allowed if we have `[12:30:10, 15:45:20]` since 30 is not within `[10, 20]`. But the seconds only matter if the current time was within `[minTime, maxTime]`.